app/services/EuTaxRegistrationValidations.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import cats.implicits._
+import models.domain.EuTaxIdentifierType.{Other, Vat}
+import models.domain._
+import models.{Country, CountryWithValidationDetails, DataMissingError, Index, InternationalAddress, UserAnswers, ValidationError, ValidationResult}
+import pages.euDetails._
+import queries.euDetails.AllEuDetailsRawQuery
+
+trait EuTaxRegistrationValidations {
+
+
+  def getEuTaxRegistrations(answers: UserAnswers): ValidationResult[List[EuTaxRegistration]] = {
+    answers.get(TaxRegisteredInEuPage) match {
+      case Some(true) =>
+        answers.get(AllEuDetailsRawQuery) match {
+          case None =>
+            DataMissingError(AllEuDetailsRawQuery).invalidNec
+          case Some(euDetails) =>
+            euDetails.value.zipWithIndex.map {
+              case (_, index) =>
+                processEuDetail(answers, Index(index))
+            }.toList.sequence
+        }
+
+      case Some(false) =>
+        answers.get(AllEuDetailsRawQuery) match {
+          case Some(_) => DataMissingError(AllEuDetailsRawQuery).invalidNec
+          case None => List.empty.validNec
+        }
+
+      case None =>
+        DataMissingError(TaxRegisteredInEuPage).invalidNec
+    }
+  }
+
+  private def processEuDetail(answers: UserAnswers, index: Index): ValidationResult[EuTaxRegistration] = {
+    answers.get(EuCountryPage(index)) match {
+      case Some(country) => getRegistrationWithFixedEstablishment(answers, country, index) // TODO: check logic here
+      case None => DataMissingError(EuCountryPage(index)).invalidNec
+    }
+  }
```

Comment:
Double check the logic. Here's the same method in OSS https://github.com/hmrc/one-stop-shop-registration-frontend/blob/8d25bc6080d13845bcdc4f4b29db0116926b5874/app/services/EuTaxRegistrationValidations.scala#L54-L67 , but we don't have `SellsGoodsToEUConsumersPage` in IOSS